File: PrimeDirective.java
// Project from https://www.codecademy.com/projects/practice/java-prime-directive

class PrimeDirective {
  
  // Check for number prime or not
    static boolean isPrime(int n)
    {

        // Check if number is less than or equal to 1
        if (n <= 1)
          return false;

        // Check if number is 2
        else if (n == 2)
          return true;

        // Check if n is a multiple of 2
        else if (n % 2 == 0)
          return false;

        // If not, then just check the odds
        for (int i = 3; i <= Math.sqrt(n); i += 2) {
          if (n % i == 0)
            return false;
        }
        return true;
    }
    
  public static void main(String[] args) {

    PrimeDirective pd = new PrimeDirective();
    int[] numbers = {6, 29, 28, 33, 11, 100, 101, 43, 89};
    for (int number: numbers){
      System.out.print(number + " is prime: " + isPrime(number));
    }
  }  

}

File: PrimeDirective.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include <stdbool.h>

bool isPrime(int number);
int main(int argc, const char *argv[])
{
  for (int i = 2; i < 13; i++)
  {
    printf("%d is prime: %d\n", i, isPrime(i));
  }
  return 0;
}

bool isPrime(int number)
{
  // anything less than or equal to one is non-prime
  if (number <= 1)
  {
    return false;
  }
  // 2 is a special case prime
  else if (number == 2)
  {
    return true;
  }
  // all even number greater than 2 are composite
  else if (number % 2 == 0)
  {
    return false;
  }

  // check all the odd values from 3 to root(number)
  for (int i = 3; i <= sqrt(number); i += 2)
  {
    if (number % i == 0)
    {
      return false;
    }
  }
  return true;
}